tools/prologs-epilogs/sudo-oslogin

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# On Google Cloud VMs that have OS Login enabled, this script will query the
+# OS Login API and add users as sudoers if they have the OS Admin Login role.
+# The sudoers entry is added at job start and removed upon job completion.
+
+# ensure that older Slurm NSS configurations are removed
+sed -i 's,slurm ,,' /etc/nsswitch.conf
+
+# only run this script if OS Login is enabled
+i_config=$(curl -s -H "Metadata-Flavor: Google" \
+	"http://metadata.google.internal/computeMetadata/v1/instance/attributes/?recursive=true" |
+	jq -r '."enable-oslogin"')
+
+p_config=$(curl -s -H "Metadata-Flavor: Google" \
+	"http://metadata.google.internal/computeMetadata/v1/project/attributes/?recursive=true" |
+	jq -r '."enable-oslogin"')
+if ! [[ "$i_config" == "TRUE" || "$p_config" == "TRUE" && "$i_config" != "FALSE" ]]; then
+	exit 0
+fi
+
+SUDOERDIR="/var/google-sudoers.d"
+JOBSUDOERFILE="${SUDOERDIR}/${SLURM_JOB_USER}-${SLURM_JOB_ID}"
+if [[ "$SLURM_SCRIPT_CONTEXT" == "prolog_slurmd" &&
+	-n "$SLURM_JOB_USER" &&
+	-n "$SLURM_JOB_ID" &&
+	-d "$SUDOERDIR" ]]; then
+
+	USER_OPENID=$(curl -s -H "Metadata-Flavor: Google" \
+		"http://metadata.google.internal/computeMetadata/v1/oslogin/users?username=${SLURM_JOB_USER}" |
+		jq -r '.loginProfiles | .[] | .name')
+
+	curl -s -H "Metadata-Flavor: Google" \
+		"http://metadata.google.internal/computeMetadata/v1/oslogin/authorize?policy=login&email=${USER_OPENID}" |
```

Comment:
To match documentation above it should be policy=adminLogin

Reply:
Good catch! Done.